hidden-posts.php

```diff
@@ -117,6 +123,35 @@ static function remove_post( $id ) {
         array_splice( $posts, array_search( $id, $posts ), 1 );
 
         update_option( self::META_KEY, array_map( 'intval', $posts ) );
+
+        // Delete post meta.
+        delete_post_meta( $id, self::META_KEY );
+    }
+		
+    /**
+     * Migrate IDs of hidden posts by duplicating them from wp_options to 
+     * wp_postmeta in case the user is using an older version of the plugin.
+     */
+    public function maybe_upgrade_version() {
+      $version = absint( get_option( self::VERSION_KEY, 0 ) );
+
+      if ( $version === self::VERSION ) {
```

Comment:
Here I think it's still better to use this operator `<` just in case the plugn needs to update other things in the future.

Reply:
I know what you mean. However using `<` would not lead to the expected result.

This would not trigger the migration script:

```
if ( $version < self::VERSION ) {
  return;
}
```

This would trigger the migration script, but if the version number would be updated with every release, the migration would also run with every release:

```
if ( $version < self::VERSION ) {

  // Place the entire migration script inside the if-statement

}
```

I'm also not a big fan of placing a huge code block inside an if-statement. I might have to improve the logic in a way that it takes the version numbers into account.

```
const MIGRATION_WP_POSTMETA = 1;
const MIGRATION_WP_USERS = 2;

if ( $version < self::MIGRATION_WP_POSTMETA ) {
  // do something
}

if ( $version < self::MIGRATION_WP_POSTMETA ) {
  // do something else
}
```

This way, I can ensure that that I can run migrations in the future and also that only the migrations that are needed are running. What do you think about that approach?

Reply:
> This would not trigger the migration script:
> 
> ```
> if ( $version < self::VERSION ) {
>   return;
> }
> ```

Haha, my brain did not think this is correct. You're right. But what's about using `>=`? I think it's good enough. But note that this is more than a nitpicking rather than anything serious. 

> This would trigger the migration script, but if the version number would be updated with every release, the migration would also run with every release:

I think we should not just increase the version here in case we want to do any migration. Otherwise, we can keep this version as-is. 

> I'm also not a big fan of placing a huge code block inside an if-statement. I might have to improve the logic in a way that it takes the version numbers into account.

Yeah, agree, it's good to avoid a huge code block in an in-statement. This improvement looks good but I think we do not need to make it complicated right now. 

In the future, we need to run other immigrations, we can implement something you're suggesting here.